oic.r.roles.raml

```diff
@@ -0,0 +1,128 @@
+#%RAML 0.8
+title: OICSecurityRolesResource
+version: v1.0-20170323
+schemas:
+  - Cred: !include schemas/oic.r.cred.json
+
+traits:
+  - interface:
+      queryParameters:
+        if:
+          enum: ["oic.if.baseline"]
+
+/oic/sec/roles:
+  description: |
+    This resource specifies roles that have been asserted.
+
+  displayName: Asserted Roles Data
+  is: [ interface ]
+
+  get:
+    description: |
+      Retrieves the asserted roles data.
+    responses:
+      200:
+        body:
+          application/json:
+            schema: Roles
+            example: |
+              {
+                "roles" :[ #array of oic.sec.cred objects
+                    {
+                      "credid":"1",
+                      "credtype":"8",  # Cred type 8 is SIGNED_ASYMMETRIC_KEY
+                      "subjectuuid":"00000000-0000-0000-0000-000000000000",
+                      "pbData":       # Role cert, type oic.sec.pubdatatype
+                        {
+                           "encoding":"oic.sec.encoding.pem",
+                           "data":"PEMENCODEDROLECERT" 
+                        }, 
+                      "optData":        # Optional issuer certificate, type oic.sec.pubdatatype
+                        {
+                           "encoding":"oic.sec.encoding.pem",
+                           "data":"PEMENCODEDISSUERCERT" 
+                        } 
+                    },
+                    {
+                      "credid":"2",
+                      "credtype":"8",  # Cred type 8 is SIGNED_ASYMMETRIC_KEY
+                      "subjectuuid":"00000000-0000-0000-0000-000000000000",
+                      "pbData":         # Role cert, type oic.sec.pubdatatype
+                        {
+                           "encoding":"oic.sec.encoding.pem",
+                           "data":"PEMENCODEDROLECERT" 
+                        }, 
+                      "optData":        # Optional issuer certificate, type oic.sec.pubdatatype
+                        {
+                           "encoding":"oic.sec.encoding.pem",
+                           "data":"PEMENCODEDISSUERCERT" 
+                        }
+                    }
+                ], 
+                "rt":["oic.sec.cred"], 
+                "if":"oic.if.baseline"
+                 }
+              }
+      400:
+        description: The request is invalid.
+
+  post:
+    description: |
+      Update the roles resource, i.e., assert new roles to this server.
```

Comment:
Is the POST a complete replacement of the Server's existing/active roles? Or are the semantics the same as for /oic/sec/cred? 

For example, the cred description for POST states:

>       Updates the credential resource with the provided credentials.
> 
>       Credentials provided in the update with credid(s) not currently in the
>       credential resource are added.
> 
>       Credentials provided in the update with credid(s) already in the
>       credential resource completely replace the creds in the credential
>       resource.
> 
>       Credentials provided in the update without credid(s) properties are
>       added and assigned unique credid(s) in the credential resource.

I would think we'd want consistent semantics for any operation that takes a list of creds. But I could be convinced otherwise.